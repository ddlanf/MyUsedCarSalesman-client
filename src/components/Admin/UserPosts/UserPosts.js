import React, { Component } from 'react'
import PostApiService from '../../../services/posts-api-service'
export default class UserPosts extends Component {

    findUserId(post){
        const { users } = this.props
        for(let i = 0; i < users.length; i++){
            if(users[i].user_name === post.user_name){
                return users[i].id
            }
        } 
    }

    deletePost = (post_id) =>{
        PostApiService.deletePostAdmin(post_id)
            .then(() => {
                this.props.updatePosts()
            })
    }

    renderUserPosts(){
        const { posts } = this.props
        const userTable = posts.map(post => {
            return (
                <tr 
                    key={post.id}
                    >
                    <th>{post.id}</th>
                    <th>{post.make}</th>
                    <th>{post.model}</th>
                    <th>{post.year}</th>
                    <th>{post.mileage}</th>
                    <th className="admin-paragraph">{post.description}</th>
                    <th>${post.price}</th>
                    <th className="admin-paragraph">{post.commission_amount}</th>
                    <th>{post.location}</th>
                    <th className="admin-paragraph">{post.other_terms_and_conditions}</th>
                    <th>{this.findUserId(post)}</th>
                    <th>
                        <button 
                             onClick={() => this.deletePost(post.id)}
                            className="admin-delete">
                            Delete
                        </button>
                    </th>
                </tr>
            )
        })

        return userTable;
    }

    
    render() {
        return (
            <div className="users-info posts">
                <h2>User Posts</h2>
                <table className="admin-table posts">
                    <thead>
                        <tr className="about-posts">
                            <th>ID</th>
                            <th>Make</th>
                            <th>Model</th>
                            <th>Year</th>
                            <th>Mileage</th>
                            <th>Description</th>
                            <th>Price</th>
                            <th>Commission Amount</th>
                            <th>Location</th>
                            <th>Other terms and conditions</th>
                            <th>User ID</th>
                            <th></th>
                        </tr>
                    </thead>
                    <tbody>
                        {this.renderUserPosts()}
                    </tbody>
                </table>
            </div>
        )
    }
}
