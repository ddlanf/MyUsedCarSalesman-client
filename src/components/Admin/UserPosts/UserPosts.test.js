import React from 'react';
import ReactDOM from 'react-dom';
import UserPosts from './UserPosts';
import renderer from 'react-test-renderer';

describe('UserPosts component', () => {
    
    it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<UserPosts
        users={[]} 
        posts={[]}
        updatePosts={function(){}}
        />, div);
    ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI as expected', () => {
        const tree = renderer
          .create(<UserPosts
            users={[]} 
            posts={[]}
            updatePosts={function(){}}
            />)
          .toJSON();
        expect(tree).toMatchSnapshot();  
      });
});