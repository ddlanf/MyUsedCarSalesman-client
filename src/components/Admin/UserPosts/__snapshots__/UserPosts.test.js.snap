// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserPosts component renders the UI as expected 1`] = `
<div
  className="users-info posts"
>
  <h2>
    User Posts
  </h2>
  <table
    className="admin-table posts"
  >
    <thead>
      <tr
        className="about-posts"
      >
        <th>
          ID
        </th>
        <th>
          Make
        </th>
        <th>
          Model
        </th>
        <th>
          Year
        </th>
        <th>
          Mileage
        </th>
        <th>
          Description
        </th>
        <th>
          Price
        </th>
        <th>
          Commission Amount
        </th>
        <th>
          Location
        </th>
        <th>
          Other terms and conditions
        </th>
        <th>
          User ID
        </th>
        <th />
      </tr>
    </thead>
    <tbody />
  </table>
</div>
`;
